### PR TITLE
Download artifact to current working folder if there is no `dest_dir` argument.

### LINKED ISSUE #4: Make the destination folder optional
If a destination folder is not provided, then we could use the current folder (like wget).

### PR DESCRIPTION
**Pull request by askeing**
fix #4

### CODE REVIEW

**Comment by askeing** [comment]
@JohanLorenzo @shakoho r?

**Comment by ShakoHo** [comment]
verified in my local environment.. r+

File: taskcluster_util/taskcluster_download.py [hunk 1]
```diff
@@ -93,19 +93,19 @@ def run(self):
...
-            print('### Please specify the artifact name and dest folder at the same time.\n')
-            exit(-1)
         else:
-            # has artifact_name and dest_dir, then download it
-            abs_dest_dir = os.path.abspath(self.options.dest_dir)
-            if os.path.exists(abs_dest_dir) and (not os.path.isdir(abs_dest_dir)):
-                print('### {} is not a folder.\n'.format(abs_dest_dir))
-                exit(-1)
+            # has artifact_name, then download it
+            if self.options.dest_dir:
```
Comment (JohanLorenzo) [inline taskcluster_util/taskcluster_download.py]: I think we can extract this if/else into a method. So the `run()` method will be smaller :)

``` python

class NotAFolderException(Exception):
    def __init__(self, path):
        super(Exception, self).__init__()
        self.message = '{} is not a folder.'.format(path)

class DownloadRunner(object):
    def _get_output_directory_path(self):
        abs_dest_dir = os.path.abspath(self.options.dest_dir) if self.options.dest_dir else os.getcwd()
        if os.path.exists(abs_dest_dir) and (not os.path.isdir(abs_dest_dir)):
            raise NotAFolderException(abs_dest_dir) 
        return abs_dest_dir

    def run(self):
        ...
        if self.options.aritfact_name is None:
            # no artifact_name, then get the latest artifacts list
            self.show_latest_artifacts(artifact_downloader, task_id)
        else:
            # has artifact_name, then download it
            abs_dest_dir = self._get_output_directory_path()
            print('### The destination folder is [{}]'.format(abs_dest_dir))
        ...
```

What do you think?

File: README.rst [hunk 2]
```diff
@@ -18,8 +18,8 @@ taskcluster_download
 
     optional arguments:
       -h, --help            show this help message and exit
-      --credential CREDENTIAL
-                            The credential JSON file (default: tc_credential.json)
+      --credentials CREDENTIALS
```
Comment (JohanLorenzo) [inline README.rst]: I don't think the modifications in this file should belong in the commit called `Download artifact to current working folder if there is no`dest_dir`argument.`

Comment (askeing) [inline README.rst]: oh! i forgot to stash before create a branch. thx you :)

File: taskcluster_util/taskcluster_download.py (hunk 1 above)
Comment (askeing) [inline taskcluster_util/taskcluster_download.py]: that's a great idea!

but I think we can just print a warning message and exit the program, due to this class is a program for user, not a library for developer.
IMO, the program for user should handle the exception, and library for developer should raise exception.
What do you think?

**Comment by askeing** [comment]
- remove the `README.rst` change
- extract the `_get_output_directory_path`

**Comment by askeing** [comment]
@ShakoHo , could you please help to take a look?
If it works fine, then we can merge it.

File: taskcluster_util/taskcluster_download.py [hunk 3]
```diff
@@ -60,6 +60,13 @@ def show_latest_artifacts(self, artifact_downloader, task_id):
         for artifact in artifacts_list:
             print('### {}| {}'.format(artifact.get('contentType').ljust(width), artifact.get('name').ljust(width)))
 
+    def _get_output_directory_path(self):
+        abs_dest_dir = os.path.abspath(self.options.dest_dir) if self.options.dest_dir else os.getcwd()
```
Comment (ShakoHo) [inline taskcluster_util/taskcluster_download.py:64]: hi Askeing, do we need to consider the write permission of given path? ..if yes, we could use os.access(path_name ,os.W_OK) to check ..

Comment (askeing) [inline taskcluster_util/taskcluster_download.py:64]: add `elif not os.access(abs_dest_dir, os.W_OK):` for this case.
thx

**Comment by JohanLorenzo** [comment]
> IMO, the program for user should handle the exception, and library for developer should raise exception. What do you think?

You're right. Exceptions should not be shown to CLI users. That said, I think there is some value in doing `exit()` in only one function of the CLI => One function will be responsible of 1 thing. So I think we can raise exception every time there is an error and catch them in `main()`:

``` python
def main():
    try:
        DownloadRunner().run()
    except Exception, e:
        logger.error(e.message)
        exit(1)
```

If you wish, we can do that in a following PR.

**Comment by askeing** [comment]
yap, great idea!
Let's do it :+1:

**Pull request merged**